src/Language/PureScript/TypeChecker/Entailment.hs

```diff
@@ -262,15 +262,19 @@ entails SolverOptions{..} constraint context hints =
                            sortOn (tcdChain &&& tcdIndex)
                            dicts
                   -- process instances in a chain in index order
-                  let found = for (init $ tails chain) $ \(tcd:tl) ->
-                                -- Make sure the type unifies with the type in the type instance definition
-                                case matches typeClassDependencies tcd tys'' of
-                                  Apart        -> Right ()                   -- keep searching
-                                  Match substs -> Left (Right (substs, tcd)) -- found a match
-                                  Unknown ->
-                                    if null (tcdChain tcd) || null tl
-                                    then Right ()                                   -- need proof of apartness but this is either not in a chain or at the end
-                                    else Left (Left (tcdToInstanceDescription tcd)) -- can't continue with this chain yet, need proof of apartness
+                  let found = for (init $ tails chain) $ \tail' ->
+                                case tail' of
```

Comment:
This is another `-Wincomplete-uni-patterns` one but I wonder if it might be nicer to use the NonEmpty version of `groupBy` above, so that we wouldn't need a branch that calls `internalError` here.